cmd/serverless-init/main.go

```diff
@@ -8,108 +8,126 @@
+	"github.com/DataDog/datadog-agent/comp/core/secrets"
+	"github.com/DataDog/datadog-agent/comp/core/tagger"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/taggerimpl"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	workloadmetafx "github.com/DataDog/datadog-agent/comp/core/workloadmeta/fx"
+	"github.com/DataDog/datadog-agent/pkg/serverless"
+
+	"go.uber.org/atomic"
 	"go.uber.org/fx"
 
 	"github.com/DataDog/datadog-agent/cmd/serverless-init/cloudservice"
-	"github.com/DataDog/datadog-agent/cmd/serverless-init/initcontainer"
-	"github.com/DataDog/datadog-agent/cmd/serverless-init/log"
 	"github.com/DataDog/datadog-agent/cmd/serverless-init/metric"
-	"github.com/DataDog/datadog-agent/cmd/serverless-init/tag"
+	serverlessInitTag "github.com/DataDog/datadog-agent/cmd/serverless-init/tag"
 	logsAgent "github.com/DataDog/datadog-agent/comp/logs/agent"
-	"github.com/DataDog/datadog-agent/pkg/config"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/config/model"
 	configUtils "github.com/DataDog/datadog-agent/pkg/config/utils"
 	"github.com/DataDog/datadog-agent/pkg/serverless/metrics"
 	"github.com/DataDog/datadog-agent/pkg/serverless/otlp"
 	"github.com/DataDog/datadog-agent/pkg/serverless/random"
-	"github.com/DataDog/datadog-agent/pkg/serverless/tags"
+	serverlessTag "github.com/DataDog/datadog-agent/pkg/serverless/tags"
 	"github.com/DataDog/datadog-agent/pkg/serverless/trace"
 	tracelog "github.com/DataDog/datadog-agent/pkg/trace/log"
 	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
-	logger "github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-const (
-	datadogConfigPath = "/var/task/datadog.yaml"
-	logLevelEnvVar    = "DD_LOG_LEVEL"
-	loggerName        = "SERVERLESS_INIT"
-)
+const datadogConfigPath = "datadog.yaml"
 
-type cliParams struct {
-	args []string
-}
+var modeConf mode.Conf
 
 func main() {
-	if len(os.Args) < 2 {
-		panic("[datadog init process] invalid argument count, did you forget to set CMD ?")
-	}
 
-	cliParams := &cliParams{
-		args: os.Args[1:],
-	}
-
-	err := fxutil.OneShot(run, fx.Supply(cliParams))
+	modeConf = mode.DetectMode()
+	setEnvWithoutOverride(modeConf.EnvDefaults)
+
+	err := fxutil.OneShot(
+		run,
+		autodiscoveryimpl.Module(),
+		workloadmetafx.Module(),
+		fx.Provide(func(config coreconfig.Component) healthprobeDef.Options {
+			return healthprobeDef.Options{
+				Port:           config.GetInt("health_port"),
+				LogsGoroutines: config.GetBool("log_all_goroutines_when_unhealthy"),
+			}
+		}),
+		taggerimpl.Module(),
+		healthprobeFx.Module(),
+		fx.Supply(workloadmeta.NewParams()),
+		fx.Supply(tagger.NewTaggerParams()),
+		fx.Supply(core.BundleParams{
+			ConfigParams: coreconfig.NewParams("", coreconfig.WithConfigMissingOK(true)),
+			SecretParams: secrets.NewEnabledParams(),
+			LogParams:    logimpl.ForOneShot(modeConf.LoggerName, "off", true)}),
+		core.Bundle(),
```

Comment:
The core bundle is overkill for the serverless binary. 

It brings useful components and at the same time a few other components that do not make sense in the serverless environment, such as the telemetry component, the system probe config component. 

I think for this case rather than using the bundle, we should include manually the component that we need in the serverless init context. Secrets, Logs, and Config 


```go
fx.Supply(coreconfig.NewParams("", coreconfig.WithConfigMissingOK(true))),
coreconfig.Module(),
fx.Supply(secrets.NewEnabledParams()),
secretsimpl.Module(),
fx.Supply(logimpl.ForOneShot(modeConf.LoggerName, "off", true)),
logimpl.Module(),
```